src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -275,5 +327,35 @@ private void AssignTimeout(long ticks, TimeoutAction timeoutAction)
             // Add Heartbeat.Interval since this can be called right before the next heartbeat.
             Interlocked.Exchange(ref _timeoutTimestamp, _lastTimestamp + ticks + Heartbeat.Interval.Ticks);
         }
+
+        public void StartTimingReads()
+        {
+            Interlocked.Exchange(ref _readTimingElapsed, 0);
+            Interlocked.Exchange(ref _bytesRead, 0);
+            Interlocked.Exchange(ref _readTimingStartTicks, _lastTimestamp);
```

Comment:
Yeah these are unnecessary. Setting `_timingReads` last should help since the other properties won't be read until that is true.